include/xsimd/xsimd.hpp

```diff
@@ -18,6 +18,14 @@
 #define XSIMD_NO_DISCARD
 #endif
 
+#if defined(__GNUC__)
+#define XSIMD_INLINE inline __attribute__((always_inline))
+#elif defined(_WIN32)
+#define XSIMD_INLINE inline __forceinline
+#else
+#define XSIMD_INLINE inline
+#endif
```

Comment:
`_WIN32` is probably the wrong thing to check for `__forceinline` support since compilers will define that for simply indicating a Windows target (and even some misguided developers will define for general windows compatibility reasons), regardless of whether support for `__forceinline` is available. An MSVC-specific macro, like `_MSC_VER`, is probably what you want to check for instead.

Additionally you may want to guard this whole construct inside an `#ifndef XSIMD_INLINE` block, as some people with more specialized needs may not want everything force-inlined and will want to override this behaviour...

Reply:
> An MSVC-specific macro, like `_MSC_VER`, is probably what you want to check for instead.

This is correct, MSYS2-based compilers just work by accident-- because they get checked first.

> Additionally you may want to guard this whole construct inside an #ifndef XSIMD_INLINE block, as some people with more specialized needs may not want everything force-inlined and will want to override this behaviour...

In the case of GCC and Clang, forcing the inlining isn't even necessary, it is automatically done on `-O2` and higher. So you may just force it on MSVC.